### DISCUSSION TITLE
Want to outline a particular hex in Bevy

### DISCUSSION THREAD

**Discussion by targrub** [ManevilleF/hexx#150] [Q&A]
How would I construct the mesh to be able to draw an outline around a chosen hex (say, in the hex_grid sample)?

**Comment by ManevilleF** [comment]
Usually outlines are part of shader code, I'm not sure what you intend to do

**Comment by targrub** [comment]
I guess I'm hoping to have a similar type to PlaneMeshBuilder that could be used, like hexagonal_plane in the hex_grid sample, to generate a mesh (basically two hexagonal shapes forming a donut with the inner radius the radius of the PlaneMeshBuilder hex, the outer radius one that reaches the adjoining PlaneMeshBuilder hex) that would form a shape around the specified hex.

**Comment by ManevilleF** [comment]
That is something I might do in the future, for now I suggest looking into the `PlaneMeshBuilder` code and see how it uses `HexLayout::hex_corners` to figure out the the vertx positions, because what you want is basically this, twice, and reconstruct the triangles

**Reply by targrub** [reply]
Thanks.  If I'm happy with my results, I could submit a PR.

**Reply by targrub** [reply]
Well, I'm happy with the results, but the code I wrote shouldn't reach the light of day.

I'm having an issue running `cargo test` on hexx, though.  
```
error: linking with `link.exe` failed: exit code: 1189
  = note: "C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.33.31629\\bin\\HostX64\\x64\\link.exe"

...

  = note: LINK : fatal error LNK1189: library limit of 65535 objects exceeded
error: could not compile `bevy_dylib` (lib) due to 1 previous error
```
I'm compiling on Windows 11, branched off of 0.14.0 version of hexx.

**Reply by ManevilleF** [reply]
Try disabling the bevy dynamic linking feature

**Reply by targrub** [reply]
Yes, that worked.

**Comment by ManevilleF** [comment]
Did #152 fully address this ?

**Reply by targrub** [reply]
I have not written a version of my code using the new code.

It would be nice to have an example of a hex grid which used the outline feature as a way to highlight the hex the cursor is over, and changed the color of a hex if it is clicked.

**Reply by ManevilleF** [reply]
I added the `hex_area` example, the two cursors use an insetted mesh if that can help understanding the API

**Reply by targrub** [reply]
That looks helpful.  Thanks!
